src/runtime_pthread.js

```diff
@@ -31,14 +31,9 @@ if (ENVIRONMENT_IS_PTHREAD) {
 
     Object.assign(globalThis, {
       self: global,
-      // Dummy importScripts.  The presence of this global is used
-      // to detect that we are running on a Worker.
-      // TODO(sbc): Find another way?
-      importScripts: () => {
```

Comment:
I'm a little confused because you say that deno never defines `importScripts`, but from the looks of this code it seems that node doesn't define it either.   Isn't the point of this line to inject it?

Reply:
Ah, you're right. Node.js doesn't define `importScripts` either.
```js
const { Worker, isMainThread } = require('node:worker_threads');

if (isMainThread) {
  // This re-loads the current file inside a Worker instance.
  new Worker(__filename);
} else {
  console.log(typeof importScripts);  // Prints 'undefined'.
}
```

Commit 4bab494ba88b664241bbff2b4427d5b24ed9e16a removes this altogether, since `ENVIRONMENT_IS_WORKER` no longer depends on this dummy symbol on Node.js:
https://github.com/emscripten-core/emscripten/blob/9e618aaaf4bfec941f04d909549ef76d880e2200/src/shell.js#L144

(I can separate this change into its own PR if needed)

Reply:
Note that `ENVIRONMENT_IS_NODE` is `false` on Deno. IIUC, Deno tends to favor standard Web Platform APIs over proprietary ones and wasn't originally designed to be a drop-in replacement for Node.js.

Reply:
Deno does have a [Node.js compatibility layer](https://docs.deno.com/runtime/fundamentals/node/) these days (they started to compromise on that at some point, though they did start with "pure Web APIs" IIRC). I wonder if maybe we can make `IS_NODE` also accept deno, for simplicity? Separate from this PR of course. (And if it isn't trivial, then https://github.com/emscripten-core/emscripten/issues/12203 is probably the best long-term plan instead.)

This PR itself lgtm as is. Unless @sbc100 you had more thoughts?